designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -49,15 +57,21 @@ object UsageStatisticsReportsSettingsDeterminer {
           GetScenarioWithDetailsOptions.detailsOnly.withFetchState
         )(user)
         .map { scenariosDetails =>
-          scenariosDetails.map(scenario =>
-            ScenarioStatisticsInputData(
-              scenario.isFragment,
-              scenario.processingMode,
-              deploymentManagerTypeByProcessingType(scenario.processingType),
-              scenario.state.map(_.status)
+          Right(
+            scenariosDetails.map(scenario =>
+              ScenarioStatisticsInputData(
+                scenario.isFragment,
+                scenario.processingMode,
+                deploymentManagerTypeByProcessingType(scenario.processingType),
+                scenario.state.map(_.status)
+              )
             )
           )
         }
+        .recoverWith { case ex: SQLException =>
```

Comment:
IMO at this level we should not deal with SQLExceptions explicitly. IMO this case should be handled at the repository implementation level.

Reply:
It's implemented deep in the old repository, should I provide alternative implementation with the error handling or for this case we can leave as it is and handle the exception here or maybe should we ignore that some exceptions can bubble in Future.failed?

Reply:
IMO there is no sense to consider all possible problems here. We should only consider the things we can predict. For eg. no scenarios in the returned list. But I don't think we should bother that eg. there won't be a connection to handle our case. We cannot do much with it. So, it's better to forget all these things and let the upper-level handler do its job (here: caching the exception, logging, and returning HTTP 500 - I guess these things are done now ;))

Reply:
Done